NotifyMe/utils/exceptionManager.py

```diff
@@ -0,0 +1,68 @@
+
+# proper response format-> { status code, a message, an actual data, status code, 
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+
+class CustomException(Exception): # CUSTOM EXCEPTION
+  def __init__(self, message, data, status_code):
+    self.message = message
+    self.data = str(data)
+    self.status_code = status_code
+    super.__init__(self.message)
+    
+    self.return_message = {
+      "message":self.message,
+      "data":str(self.data),
+      "status":self.status_code
+    }
+      
+    
+    def __str__(self):
+      return self.return_message
+    
+  
+class SuccessHandler:
+  
+  @staticmethod
+  def handle_success(message, data=None, status_code=200):
+    """
+    For sending success message...
+    """
+    logger.info(f"Success: {message} (status: {status_code})") 
+    
+    response_data = {
+      "message":message,
+      "data":data,
+      "status":status_code
+    }
+    return response_data
+    
+    
+class ExceptionHandler:
+    
+    @staticmethod
+    def handle_exception(message, data=None, status_code=500):
+      """
+      for normal error handling in service and serializer.
+    
+      """
+      logger.error(f"Exception: {message} (status: {status_code})")
+      raise CustomException(message, data, status_code)
+    
+    
+    @staticmethod
+    def handle_api_exception(message, data=None, status_code=500):
```

Comment:
for api exception why are raising the exception instead return in the response

Reply:
I have to raise the error then only i can catch the error that will be returned in the response